metrics/ci_integrations/test/cli/logger/logger_test.dart

```diff
@@ -1,65 +1,197 @@
+      () {
+        const sourceClass = _ClassStub;
+        final expectedLogger = Logger(
+          sourceClass: sourceClass,
+          writer: writerMock,
+          verbose: testVerbose,
+        );
+
+        when(loggerFactoryMock.create(sourceClass)).thenReturn(expectedLogger);
+        final logger = Logger.forClass(sourceClass);
+        final secondLogger = Logger.forClass(sourceClass);
+
+        expect(logger, isNotNull);
+        expect(logger.sourceClass, equals(sourceClass));
+        expect(secondLogger, same(logger));
+        verify(loggerFactoryMock.create(sourceClass)).called(1);
+      },
+    );
+
+    test(
+      ".message() logs the given message using the specified logger writer",
       () {
-        Logger.setup(messageSink: sinkMock);
-        Logger.logMessage(message);
+        const message = 'test';
+
+        logger.message(message);
 
-        verify(sinkMock.writeln(message)).called(1);
+        verify(writerMock.write(message)).called(1);
       },
     );
 
     test(
-      ".logInfo() logs the given message to the message sink if the verbose is true",
+      ".message() logs the given message with timestamp and source class prefixes if in the verbose mode",
       () {
-        final dateTimeNow = DateFormat.yMd().add_Hms().format(DateTime.now());
-        final expected = '[$dateTimeNow] $message';
+        const message = 'test';
 
-        Logger.setup(messageSink: sinkMock, verbose: true);
-        Logger.logInfo(message);
+        verboseLogger.message(message);
+        final logMatcher = predicate<String>(
+          (log) => matchesPrefixes(log, '$testSourceClass', message),
+        );
 
-        verify(sinkMock.writeln(expected)).called(1);
+        verify(writerMock.write(argThat(logMatcher))).called(1);
       },
     );
 
     test(
-      ".logInfo() does not log the given message to the message sink if the verbose is false",
+      ".info() does not write log if logger is not in the verbose mode",
       () {
-        Logger.setup(messageSink: sinkMock, verbose: false);
-        Logger.logInfo(message);
+        const message = 'test';
+
+        logger.info(message);
 
-        verifyNever(sinkMock.writeln(message));
+        verifyNever(writerMock.write(any));
+      },
+    );
+
+    test(
+      ".info() logs the given message using the specified logger writer",
+      () {
+        const message = 'test';
+
+        verboseLogger.info(message);
+
+        verify(writerMock.write(
+          argThat(contains(message)),
+        )).called(1);
+      },
+    );
+
+    test(
+      ".info() logs the given message with timestamp and source class prefixes",
+      () {
+        const message = 'test';
+
+        verboseLogger.info(message);
+        final logMatcher = predicate<String>(
+          (log) => matchesPrefixes(log, '$testSourceClass', message),
+        );
+
+        verify(writerMock.write(argThat(logMatcher))).called(1);
       },
     );
   });
 }
+
+class _ClassStub {}
```

Comment:
It seems like this class lacks documentation. Please, go through all tests and apply this suggestion.